compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -12845,195 +12845,292 @@ TR::Node *su2dSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
    }
 
 
+bool canTransformOp(bool canTransformAdd, bool canTransformSub, TR::Node *node, TR::Simplifier *s)
+   {
+   if (!(canTransformAdd || canTransformSub))
+      {
+      if (s->trace())
+         traceMsg(s->comp(),
+                  "\nEliminating add/sub under order compare node n%dn failed due to overflow\n",
+                  node->getGlobalIndex());
+
+      return false;
+      }
+   return true;
+   }
+
+// Overflow/Underflow handling:
+// Must detect overflow without causing overflow:
+// Equation: add operation: x + Const1 < Const2 --> x < Const2 - Const1
+//          sub operation: x - Const1 < Const2 --> x < Const2 + Const1
+
+//Signed Constants may be negative, therefore underflow and overflow are possible for both add and sub operations
+bool isNotSignedOverflow(bool isAddOp, bool isSubOp, uint64_t oldConst1, uint64_t oldConst2, uint64_t signedMax){
+   return (isAddOp && !(oldConst1 < 0 && (oldConst2 > signedMax + oldConst1)))
+            || (isSubOp && !(oldConst1 > 0 && (oldConst2 > signedMax - oldConst1)));
+}
+
+bool isNotSignedUnderflow(bool isAddOp, bool isSubOp, uint64_t oldConst1, uint64_t oldConst2, uint64_t signedMin){
```

Comment:
this logic looks pretty much the same as the unsigned case - is there a simplification to be made here?